tests/filesystem/test_cloud.py

```diff
@@ -0,0 +1,110 @@
+import boto3
+from tamr_toolbox.filesystem import cloud
+import pytest
+import google
+import botocore
+import tempfile
+from unittest.mock import patch
+
+
+def test_gcs_download():
+    @patch("google.cloud.client.Client")
+    def test_download(mock_client):
+        client = mock_client()
+
+        with tempfile.NamedTemporaryFile() as download_tmp:
+            cloud.gcs_download(
+                cloud_client=client,
+                source_filepath="path_in_bucket.txt",
+                destination_filepath=download_tmp.name,
+                bucket_name="test-bucket",
+            )
+
+    test_download()
```

Comment:
`patch` can be used in a more simple way by just using it on your test function. Then you don't need to have this extra function `test_download`:
```suggestion
@patch("google.cloud.client.Client")
def test_gcs_download(mock_client):
    client = mock_client()

    with tempfile.NamedTemporaryFile() as download_tmp:
        cloud.gcs_download(
            cloud_client=client,
            source_filepath="path_in_bucket.txt",
            destination_filepath=download_tmp.name,
            bucket_name="test-bucket",
        )
```